tests/test_write_to_elasticsearch_stage_pipe.py

```diff
@@ -0,0 +1,110 @@
+from morpheus.stages.output.write_to_elasticsearch_stage import WriteToElasticsearchStage
+
+
+def connection_kwargs_func(kwargs):
+    kwargs["retry_on_status"] = 3
+    kwargs["retry_on_timeout"] = 3 * 10
+
+    return kwargs
+
+
+@pytest.fixture(scope="function", name="connection_conf_file")
+def connection_conf_file_fixture(tmp_path):
+    connection_kwargs = {"hosts": [{"host": "localhost", "port": 9201, "scheme": "http"}]}
+
+    connection_conf_file = tmp_path / "connection_kwargs_conf.yaml"
+    with connection_conf_file.open(mode="w") as file:
+        yaml.dump(connection_kwargs, file)
+
+    yield connection_conf_file
+
+
+@pytest.mark.use_python
+@pytest.mark.parametrize("conf_file, exception", [("connection_conf.yaml", FileNotFoundError), (None, Exception)])
+def test_constructor_invalid_conf_file(config: Config,
+                                       conf_file: str,
+                                       exception: typing.Union[Exception, FileNotFoundError]):
+    with pytest.raises(exception):
+        WriteToElasticsearchStage(config, index="t_index", connection_conf_file=conf_file)
+
+
+@pytest.mark.use_python
+@patch("morpheus.controllers.elasticsearch_controller.Elasticsearch")
+def test_constructor_with_custom_func(config: Config, connection_conf_file: str):
+    expected_connection_kwargs = {
+        "hosts": [{
+            "host": "localhost", "port": 9201, "scheme": "http"
+        }], "retry_on_status": 3, "retry_on_timeout": 30
+    }
+
+    stage = WriteToElasticsearchStage(config,
+                                      index="t_index",
+                                      connection_conf_file=connection_conf_file,
+                                      connection_kwargs_update_func=connection_kwargs_func)
+
+    assert stage._controller._connection_kwargs == expected_connection_kwargs
+
+
+@pytest.mark.use_python
+@patch("morpheus.stages.output.write_to_elasticsearch_stage.ElasticsearchController")
+def test_write_to_elasticsearch_stage_pipe(mock_controller: typing.Any,
+                                           connection_conf_file: str,
+                                           config: Config,
+                                           filter_probs_df: typing.Union[cudf.DataFrame, pd.DataFrame]):
+    mock_df_to_parallel_bulk_write = mock_controller.return_value.df_to_parallel_bulk_write
+    mock_refresh_client = mock_controller.return_value.refresh_client
+
+    # Create a pipeline
+    pipe = Pipeline(config)
+
+    # Add the source stage and the WriteToElasticsearchStage to the pipeline
+    source_stage = pipe.add_stage(InMemorySourceStage(config, [filter_probs_df]))
+
+    write_to_elasticsearch_stage = pipe.add_stage(
+        WriteToElasticsearchStage(config, index="t_index", connection_conf_file=connection_conf_file))
+
+    # Connect the stages in the pipeline
+    pipe.add_edge(source_stage, write_to_elasticsearch_stage)
+
+    # Run the pipeline
+    pipe.run()
```

Comment:
Hint: You can use a `LinearPipeline` here which avoids the need to call `add_edge`